Features/CaseSearch/test_cases/test_03_caseseacrh_misc.py

```diff
@@ -0,0 +1,153 @@
+    webapps.select_case_and_continue("Bugs")
+    webapps.open_form("Shows")
+    webapps.omni_search("Justin")
+    casesearch.check_values_on_caselist(row_num="4", value="Justin")
+    case_name = webapps.omni_search("Bangalore")
+    webapps.select_case(case_name)
+    casesearch.check_value_on_case_detail(search_property="Parent Artist", expected_value="Justin")
+
+
+def test_case_03_auto_advance_menus(driver):
+    webapps = WebApps(driver)
+    casesearch = CaseSearchWorkflows(driver)
+    """Check auto advance to forms"""
+    webapps.open_app(CaseSearchUserInput.case_search_app_name)
+    webapps.open_menu(CaseSearchUserInput.artist_menu)
+    case_name = webapps.omni_search("Bugs Artist")
+    webapps.select_case_and_continue(case_name)
+    casesearch.check_eof_navigation(eof_nav="MENU", menu="Bugs Artist")
+
+
+def test_case_04_display_only_forms(driver):
+    webapps = WebApps(driver)
+    base = BasePage(driver)
+    """Check display only form modes"""
+    webapps.open_app(CaseSearchUserInput.case_search_app_name)
+    webapps.open_menu("Play Song - Display Only Forms")
+    assert not base.is_displayed(webapps.search_all_cases_button)
+
+
+def test_case_05_shadow_menu(driver):
+    webapps = WebApps(driver)
+    casesearch = CaseSearchWorkflows(driver)
+    """Check forms in normal menu"""
+    webapps.open_app(CaseSearchUserInput.case_search_app_name)
+    webapps.open_menu(CaseSearchUserInput.search_first_menu)
+    webapps.clear_selections_on_case_search_page()
+    casesearch.search_against_property(search_property="Song Name", input_value="Bugs", property_type="TEXT_INPUT")
+    webapps.search_button_on_case_search_page()
+    search_first_form_names = webapps.select_case_and_continue("Bugs")
+    """Check search and forms in shadow menu"""
+    webapps.open_app(CaseSearchUserInput.case_search_app_name)
+    webapps.open_menu("Shadow Menu")
+    casesearch.search_against_property(search_property="Rating", input_value="*****", property_type="COMBOBOX")
+    webapps.search_button_on_case_search_page()
+    case_name = webapps.omni_search("Kyon")
+    shadow_form_names = webapps.select_case_and_continue(case_name)
+    assert shadow_form_names == search_first_form_names
+
+
+def test_case_06_performance_check(driver):
+    webapps = WebApps(driver)
+    casesearch = CaseSearchWorkflows(driver)
+    """Check performance"""
+    webapps.open_app(CaseSearchUserInput.case_search_app_name)
+    webapps.open_menu("Musical Instruments (Performance)")
+    webapps.open_form("View Instruments")
+    webapps.search_all_cases()
+    casesearch.search_against_property(search_property="Instrument Name", input_value="Guitar",
+                                       property_type="TEXT_INPUT")
+    start_time = time.perf_counter()  # Start capturing time
+    webapps.search_button_on_case_search_page()
+    end_time = time.perf_counter()  # Stop capturing time
+    run_time = end_time - start_time
+    assert run_time <= 10
```

Comment:
Not really! I had just added a random number while writing the scripts.
[Based on these readings](https://docs.google.com/spreadsheets/d/1CMfAdM2Q4WMA3N4PLE-bWCCl72B9GNSVoT7zudD7OMc/edit#gid=1256916003), the max time it takes to return 10k data is 2 secs. Generally also noted that it takes less than 2 secs on staging (since it has less data). 

So adding **4** as an acceptable number by adding a buffer of 2 more seconds in https://github.com/dimagi/dimagi-qa/pull/156/commits/f4ffe8810a451b3f8c7ed6f0de4274643b61a912

Let me know your thoughts.